docs/groovy-script/groovy/differences_from_java.md

```diff
@@ -0,0 +1,358 @@
+:::
+
+## Return
+
+When a function expects to return a specific value, simply writing a value of that type at the end of the function will cause that to be returned.
+
+::: code-group
+
+```java
+public class DemoClass {
+    public DemoClass foo() {
+        return this;
+    }
+
+    public String hello() {
+        return "hello";
+    }
+
+    public String test(boolean test) {
+        if (test) {
+            return "Test is true";
+        } else {
+            return "Test is false";
+        }
+    }
+}
+```
+
+```groovy
+class DemoClass {
+    DemoClass foo() {
+        this
+    }
+
+    def hello() {
+        'hello'
+    }
+
+    def test(test) {
+        if (test) {
+            'Test is true'
+        } else {
+            'Test is false'
+        }
+    }
+}
+```
+
+:::
+
+
+## Strings
+
+In Java, declaring using single quotes `''` only accepts a single character and creates a `char`,
+and you must use double quotes `""` to create a normal `String`.
+
+In Groovy, there is no shorthand to declare a `char`,
+declaring using single quotes `''` creates a normal `String`,
+and using double quotes `""` creates a special object called a `GString`,
+which functions as a normal `String` object unless it contains a dollar sign `$`, in which case it has specific formatting rules.
+
+Learn more about this over on the specific [Strings](./strings.md) page.
+
+::: code-group
+
+```java
+char character = 'a';
+String name = "example";
+```
+
+```groovy
+char character0 = 'a'
+def character1  = (char) 'a'
+def character2  = 'a' as char
+String name0    = "example"
+def name1       = 'example'
+def name2       = "example"
+def nameGString = "$name0" // will create a GString, not a String, and will replace with the variable referenced // [!code warning]
+def nameInvalid = '$name0' // will have the raw text provided
+```
+
+:::
+
+
+## Omitting `.class`
+
+When you want to refer to a specific `class` object, you can omit the `.class` suffix.
+
+::: code-group
+
+```java
+public class DemoClass {
+    public Class<?> getClass() {
+        return DemoClass.class;
+    }
+}
+```
+
+```groovy
+class DemoClass {
+    def getClass() {
+        DemoClass
+    }
+}
+```
+
+:::
+
+
+## Repeating Variable Reference
+
+
+Groovy has multiple ways of simplifying interacting with multiple methods or fields of the same object.
+
+::: code-group
+
+```java
+DemoClass obj = new DemoClass();
+obj.sayHi();
+obj.value = 1;
+obj.name = 'hello';
+obj.total = 100;
+obj.clay = 'clay';
+obj.current = 5;
+obj.sayGoodbye();
+```
+
+```groovy [groovy with]
+DemoClass obj = new DemoClass()
+obj.with {
+    sayHi()
+    value = 1
+    name = 'hello'
+    total = 100
+    clay = 'clay'
+    current = 5
+    sayGoodbye() // returns the value of sayGoodbye()
+}
+```
+
+```groovy [groovy inbuilt with]
+DemoClass obj = new DemoClass().with {
+    sayHi()
+    value = 1
+    name = 'hello'
+    total = 100
+    clay = 'clay'
+    current = 5
+    sayGoodbye()
+    it // returns 'it'
```

Comment:
maybe link to closures page and explain that it is the modified object here

Reply:
isnt possible to link from within a codeblock, but i expanded the comment to indicate that this is caused by Groovy's closure rules